beacon-chain/core/state/state.go

```diff
@@ -187,6 +187,127 @@ func GenesisBeaconState(deposits []*ethpb.Deposit, genesisTime uint64, eth1Data
 	return state, nil
 }
 
+// OptimizedGenesisBeaconState is used to create a state that has already processed deposits. This is to efficiently
+// create a mainnet state at chainstart.
+func OptimizedGenesisBeaconState(genesisTime uint64, bState *pb.BeaconState, eth1Data *ethpb.Eth1Data) (*pb.BeaconState, error) {
```

Comment:
Is the idea to eventually deprecate the original genesis state func? It's a bit weird to have an optimized and non-optimized version of a function. Perhaps some refactoring could be used so we have only one of these functions and the heavy lifting is done outside of it?